packages/forms/src/components/DateInput/DateInput.stories.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { action } from "@storybook/addon-actions";
+
+import DateInput from "./DateInput";
+import Form from "../BasicForm";
+import Submit from "../Submit";
+import { DATE_SEGMENT } from "./types";
+
+export default {
+  component: DateInput,
+  title: "Form/Date Input",
+  argTypes: {
+    show: {
+      control: {
+        type: "check",
+        labels: {
+          [DATE_SEGMENT.Year]: "Year",
+          [DATE_SEGMENT.Month]: "Month",
+          [DATE_SEGMENT.Day]: "Day",
+        },
+      },
+      options: [DATE_SEGMENT.Year, DATE_SEGMENT.Month, DATE_SEGMENT.Day],
+    },
+  },
+} as ComponentMeta<typeof DateInput>;
+
+const Template: ComponentStory<typeof DateInput> = (args) => {
+  return (
+    <Form onSubmit={(data) => action("Submit Form")(data)}>
+      <DateInput {...args} />
+      <Submit />
+    </Form>
+  );
+};
+
+export const Default = Template.bind({});
+Default.args = {
+  idPrefix: "date",
+  legend: "Date",
+  name: "date",
+};
+
+export const WithinRange = Template.bind({});
+WithinRange.args = {
+  idPrefix: "date",
+  legend: "Date",
+  name: "date",
+  dateRange: {
```

Comment:
It's not possible to use the standard "rules" prop for this?
![image](https://user-images.githubusercontent.com/8978655/225395640-5b40e919-a624-4de6-94ed-b607becb8837.png)

Reply:
Hmm, that is interesting. I was under the impression `min` only worked with numbers and not dates 🤔 I'll take a look at this.

Reply:
Omg, it works! 🤣 I had no idea!

Added in the refactor: [refactor to use single value Controller](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5928/commits/2fa762ee94246c30b68f0b9c74ffa1586f627567)

Reply:
Okay, looks like it doesnt work 😢 What if we still use the `rules` but extract out `min` and `max` and use it for a custom validate similar to what we were doing before the the `dateRange`? 

That way, we can [use the validate object](https://github.com/react-hook-form/react-hook-form/discussions/8631#discussioncomment-3081122) and still keep things familiar by using `rules.min` and `rules.max`?

Reply:
Yeah 👎 Updated in [update to use custom validation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5928/commits/ca51c01cb7da5c3e50ba0320448ab3a2f680a2cb)